src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java

```diff
@@ -299,11 +300,18 @@ public final void execute() throws MojoExecutionException, MojoFailureException
             // We're skipping the signing stuff
             return;
         }
-        if (bestPractices && (isNotBlank(passphrase) || isNotBlank(passphraseServerId))) {
-            // Stop propagating worst practices: passphrase MUST NOT be in any file on disk
-            throw new MojoFailureException(
-                    "Do not store passphrase in any file (disk or SCM repository), rely on GnuPG agent or provide passphrase in "
-                            + passphraseEnvName + " environment variable.");
+        if (bestPractices) {
+            if (isNotBlank(passphrase) || isNotBlank(passphraseServerId)) {
+                // Stop propagating worst practices: passphrase MUST NOT be in any file on disk
+                throw new MojoFailureException(
+                        "Do not store passphrase in any file (disk or SCM repository), rely on GnuPG agent or provide passphrase in "
+                                + passphraseEnvName + " environment variable.");
+            }
+        } else {
```

Comment:
As best practices will probably be extended in the future, why not just refactor out a method here?

```suggestion
        if (bestPractices) {
            checkForBestPractices();
        } else {
```